gcc/rust/lex/rust-lex.cc

```diff
@@ -186,6 +188,56 @@ Lexer::skip_input ()
   skip_input (0);
 }
 
+void
+Lexer::skip_token (int n)
+{
+  // dump tokens if dump-lex option is enabled
+  if (dump_lex_out.is_some ())
+    {
+      dump_until_n (n);
+    }
+  else
+    {
+      token_queue.skip (n);
+    }
+}
+
+void
+Lexer::dump_until_n (int n)
```

Comment:
Ah, so the name I suggested does not make sense anymore since you "either dump or simply skip" in `skip_token (int n)`. So I think this function should be renamed. Something like `void Lexer::dump_and_skip (int n)`